src/dotnet-core-uninstall/Shared/VSVersioning/VSVersionHelper.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo;
+using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo.Versioning;
+using Microsoft.DotNet.Tools.Uninstall.Shared.Exceptions;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Tools.Uninstall.Shared.VSVersioning
+{
+    /// <summary>
+    /// Extracts VS versions and determine which sdk's are safe to uninstall
+    /// </summary>
+    internal static class VSVersionHelper
+    {
+        /// <summary>
+        /// Divisions within version bands. Not inclusive, groupings: [2.1.000, 2.1.600) [2.1.600, 2.2.00) [2.0.00, 2.2.200) [2.2.200, 2.3.00)
+        /// </summary>
+        private static readonly SemanticVersion[] SpecialCaseDivisions = { new SemanticVersion(2, 1, 600), new SemanticVersion(2, 2, 200) };
+
+        /// <summary>
+        /// Assigns the uninstallAllowed flag to false for the highest sdkVersion in each band (Major.Minor) to ensure vs still functions properly
+        /// Note that 2.1.5/2.1.6 and 2.2.1/2.2.2 count as band cutoffs 
+        /// </summary>
+        public static void AssignUninstallAllowed(IEnumerable<Bundle> bundles)
```

Comment:
"Assigns the uninstallAllowed flag to false for the highest sdkVersion in each band (Major.Minor) to ensure vs still functions properly" part is not necessary. It is almost a one to one translation to english of the linq expression